### PR TITLE
Add step in removing a like from a user to also remove unread notifications

### LINKED ISSUE #3151: Like not showing on solution
I got a notification that there was a new 'like' on one of my solutions:

exercism - notification screenshot

But when I visit the [Strain in Ruby link](http://exercism.io/submissions/f72e42aa0c244a33a2e9e738cf2bc349), there is no like visible:
exercism - no like information screenshot

### PR DESCRIPTION
**Pull request by mhelmetag**
Add step in removing a like from a user to also remove unread notifications for that like.

Possibly fixes #3151

Thoughts?

### CODE REVIEW

File: lib/exercism/submission.rb [hunk 1]
```diff
@@ -103,6 +103,7 @@ def like!(user)
   def unlike!(user)
     likes.where(user_id: user.id).destroy_all
     self.is_liked = liked_by.length > 0
+    notifications.unread.where(action: 'like', actor_id: user.id).destroy_all
```
Comment (Insti) [inline lib/exercism/submission.rb]: if `like!` doesn't know about notifications, `unlike!` should not know about notifications.
This suggests to me that this is the wrong place for this code.

Comment (Insti) [inline lib/exercism/submission.rb]: Also I suspect this line will delete ALL unread like notifications, not just the one that applies to this submission.

Comment (mhelmetag) [inline lib/exercism/submission.rb]: Not worries! Just curious hah. I was just super surprised. Thanks for the review. Should have those changes and tests soon.

**Comment by mhelmetag** [comment]
@Insti , moved the notification removal into the route instead of the model. Also rewrote the model test for the route.

**Comment by mhelmetag** [comment]
@Insti , thanks!

File: test/app/submissions_test.rb [hunk 2]
```diff
@@ -106,6 +106,23 @@ def test_unlike_a_submission
     post "/submissions/#{submission.key}/unlike", {}, login(bob)
   end
 
+  def test_unlike_removes_unread_likes
+    fred = User.create({username: 'bob', github_id: 3, email: "bob@example.com"})
```
Comment (mhelmetag) [inline test/app/submissions_test.rb]: This is a bit of a weird typo. I will fix this before merging...

Comment (mhelmetag) [inline test/app/submissions_test.rb]: Should be good now.

File: lib/exercism/submission.rb (hunk 1 above)
Comment (kytrinyx) [inline lib/exercism/submission.rb]: > I can also add a test to prove all notifications aren't deleted (just for some piece of mind).

That would be awesome. Deletion queries are particularly scary :)

File: test/app/submissions_test.rb (hunk 2 above)
Comment (kytrinyx) [inline test/app/submissions_test.rb]: `fred = ... username: "bob", email: "bob@example.com"` - totally not important, but want to make these _fred_ also?

Comment (kytrinyx) [inline test/app/submissions_test.rb]: Wait. That's exactly what you did. Sheesh. I should learn to read. Thank you :heart:

**Comment by kytrinyx** [comment]
Thank you @mhelmetag!

File: test/app/submissions_test.rb (hunk 2 above)
Comment (mhelmetag) [inline test/app/submissions_test.rb]: I really liked when line comments disappeared on change... Why, GitHub!?? Why???? (Not that I don't like all the other new fancy features).

**Pull request merged**